src/careamics/models/lvae/layers.py

```diff
@@ -727,33 +814,37 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def forward(self, latent: torch.Tensor, lowres: torch.Tensor) -> torch.Tensor:
-        """
+        """Forward pass.
+
         Parameters
         ----------
         latent: torch.Tensor
             The output latent tensor from previous layer in the LVAE hierarchy.
         lowres: torch.Tensor
             The low-res patch image to be merged to increase the context.
         """
+        # TODO: treat (X, Y) and Z differently (e.g., line 762)
         if self.retain_spatial_dims:
             # Pad latent tensor to match lowres tensor's shape
+            # Output.shape == Lowres.shape (== Input.shape),
+            # where Input is the input to the BU layer
             latent = pad_img_tensor(latent, lowres.shape[2:])
         else:
             # Crop lowres tensor to match latent tensor's shape
-            lh, lw = lowres.shape[-2:]
-            h = lh // self.multiscale_lowres_size_factor
-            w = lw // self.multiscale_lowres_size_factor
-            h_pad = (lh - h) // 2
-            w_pad = (lw - w) // 2
-            lowres = lowres[:, :, h_pad:-h_pad, w_pad:-w_pad]
+            lz, ly, lx = lowres.shape[2:]
+            z = lz // self.multiscale_lowres_size_factor
+            y = ly // self.multiscale_lowres_size_factor
+            x = lx // self.multiscale_lowres_size_factor
+            z_pad = (lz - z) // 2
+            y_pad = (ly - y) // 2
+            x_pad = (lx - x) // 2
+            lowres = lowres[:, :, z_pad:-z_pad, y_pad:-y_pad, x_pad:-x_pad]
```

Comment:
This might be wrong, since we don't want to touch the z dimension. 
At the same time this `else` will never be executed in standard cases (and, anyway, I solved this in another PR I am working on)